libcloud/container/drivers/rancher.py

```diff
@@ -0,0 +1,1212 @@
+        return "%s %s" % (self.code, self.message)
+
+    def __repr__(self):
+        return "RancherException %s %s" % (self.code, self.message)
+
+
+class RancherConnection(ConnectionUserAndKey):
+
+    responseCls = RancherResponse
+    timeout = 30
+
+    def add_default_headers(self, headers):
+        """
+        Add parameters that are necessary for every request
+        If user and password are specified, include a base http auth
+        header
+        """
+        headers['Content-Type'] = 'application/json'
+        headers['Accept'] = 'application/json'
+        if self.key and self.user_id:
+            user_b64 = base64.b64encode(b('%s:%s' % (self.user_id, self.key)))
+            headers['Authorization'] = 'Basic %s' % (user_b64.decode('utf-8'))
+        return headers
+
+
+class RancherContainerDriver(ContainerDriver):
+
+    type = Provider.RANCHER
+    name = 'Rancher'
+    website = 'http://rancher.com'
+    connectionCls = RancherConnection
+    # Holding off on cluster support for now.
+    # Only Environment API interaction enabled.
+    supports_clusters = False
+    # As in the /v1/
+    version = '1'
+
+    def __init__(self, key, secret, secure=False, host='localhost', port=80):
+        """
+        Rancher Container driver class.
+
+        Example:
+
+        >>> from libcloud.container.providers import get_driver
+        >>> from libcloud.container.types import Provider
+
+        >>> driver = get_driver(Provider.RANCHER)
+        >>> connection = driver(key="ACCESS_KEY_HERE",
+        secret="SECRET_KEY_HERE", host="172.30.0.100", port=8080)
+
+        >>> image = ContainerImage("hastebin", "hastebin", "rlister/hastebin",
+        "latest", driver=None)
+        >>> newcontainer = connection.deploy_container("myawesomepastebin",
+        image, environment={"STORAGE_TYPE": "file"})
+
+        :param    key: API key or username to used (required)
+        :type     key: ``str``
+
+        :param    secret: Secret password to be used (required)
+        :type     secret: ``str``
+
+        :param    secure: Whether to use HTTPS or HTTP.
+        :type     secure: ``bool``
+
+        :param    host: Override hostname used for connections.
+        :type     host: ``str``
+
+        :param    port: Override port used for connections.
+        :type     port: ``int``
+
+
+        :return: ``None``
+        """
+        super(RancherContainerDriver, self).__init__(key=key, secret=secret,
+                                                     secure=secure, host=host,
+                                                     port=port)
+        if host.startswith('https://'):
+            secure = True
+
+        # strip the prefix
+        prefixes = ['http://', 'https://']
+        for prefix in prefixes:
+            if host.startswith(prefix):
+                host = host.strip(prefix)
+
+        self.connection.host = host
+        self.connection.port = port
+        self.connection.secure = secure
+
```

Comment:
Why are these three lines necessary if the base class is suppose to initialize the connection? Seems redundant.